source/fundamentals/data-formats/pojo-customization.txt

```diff
@@ -10,78 +10,653 @@ POJO Customization
+   import org.bson.codecs.pojo.annotations.BsonRepresentation;
+
+   @BsonDiscriminator(value="AnnotatedProduct", key="_cls")
+   public class Product {
+       @BsonProperty("modelName")
+       private String name;
+
+       @BsonId()
+       @BsonRepresentation(BsonType.OBJECT_ID)
+       private String serialNumber;
+
+       @BsonIgnore
+       private List<Product> relatedItems;
+
+       @BsonCreator
+       public ProductPojo(@BsonProperty("modelName") String name) {
+           this.name = name;
+       }
+
+       // ...
+
+   }
+
+.. _bson-annotation-convention:
+
+.. note::
 
-Advanced configuration
+   When using annotations, remember to specify the
+   ``Conventions.ANNOTATION_CONVENTION`` in your ``ClassModelBuilder``
+   or ``PojoCodecProvider.Builder``. For example:
+
+   .. code-block:: java
+
+      ClassModel<Product> classModel = ClassModel.builder(Product.class).
+             conventions(Arrays.asList(Conventions.ANNOTATION_CONVENTION)).build();
+
+The annotations in the example POJO specify the following behavior:
+
+- reference the POJO with the specified discriminator key and   value, adding
+  the "_cls" field with the value of "AnnotatedProduct" to the BSON document
+  on write operations
+- convert between the POJO ``name`` field and value and the BSON ``modelName``
+  field and value in the document
+- convert between the POJO ``serialNumber`` field and value the BSON document
+  ``_id`` field and value in the document
+- omit the ``relatedItems`` field and value when converting data
+- use the ``Product(String name)`` constructor when instantiating the POJO
+
+.. _pojo-discriminators:
+
+Discriminators
+--------------
+
+A discriminator is a property that identifies a specific document schema. The
+discriminator key identifies a document field to use to identify the schema.
+The discriminator value identifies the default value of the document field.
+
+Use discriminators to instruct the ``CodecProvider`` which object class to use
+when deserializing to different object classes from the same collection. When
+serializing the POJO to a MongoDB collection, the associated Codec sets the
+discriminator key-value field, unless otherwise specified in the POJO property
+data.
+
+You can set and enable a discriminator in a POJO by performing one of the
+following:
+
+- use the ``@BsonDiscriminator`` annotation to specify the discriminator for
+  the POJO class
+- or call ``enableDiscriminator(true)`` on the ``ClassModelBuilder``
+  associated with the POJO class
+
+See the following example POJO classes that contain ``@BsonDiscriminator``
+annotations and example documents that contain the discriminator fields:
+
+.. code-block:: java
+   :caption: Sample POJO classes
+
+  @BsonDiscriminator(value="AnonymousUser", key="_cls")
+  public class AnonymousUser {
+      // class code
+  }
+
+  @BsonDiscriminator(value="RegisteredUser", key="_cls")
+  public class RegisteredUser {
+      // class code
+  }
```

Comment:
For some reason, I don't see syntax highlighting for this code sample. Not a huge problem, but a big jarring since the other examples all have syntax highlighting, which is particularly obvious for annotations.

Reply:
I'll add emphasize-lines if syntax highlighting isn't working. I think the implementation of syntax highlighting for Java is poor because it doesn't pick up anything unless wrapped in "public class" except some specific keywords. I'll probably open a DOP ticket if I can find something better.